api4/user.go

```diff
@@ -1191,6 +1193,64 @@ func generateMfaSecret(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(secret.ToJson()))
 }
 
+func generateRecovery(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.App.Session.IsOAuth {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		c.Err.DetailedError += ", attempted access by oauth app"
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(c.App.Session, c.Params.UserId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
+
+	codes, err := c.App.GenerateRecovery(c.Params.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+	jsonCodes, _ := json.Marshal(codes)
+
+	w.Header().Set("Cache-Control", "no-cache")
```

Comment:
Probably want no-store here not no-cache.
Also don't need the Expires header, we don't support IE anymore.